src/containers/MapContainer/PointClusterLayer.tsx

```diff
@@ -50,21 +53,40 @@ const PointClusterLayer: React.FC<PointClusterLayerProps> = ({ onItemSelected })
     state?.map?.layers.filter((l:any) => layerNames.current.includes(l.name) && state?.map?.legend[l.name]),
   );
 
+  const onMarkerSelected = (map: any, layer: any, onMarkerSelected: Function, isMarker: boolean) => {
+    const element = isMarker ? layer._icon : layer.getElement();
+    const bounds = isMarker ? [layer.getLatLng(), layer.getLatLng()] : layer.getBounds();
+    map.fitBounds(bounds);
+    if (isMarker) {
+      map.setZoom(16);
+    }
+
+    onMarkerSelected(isMarker ? 'devices' : 'cameras', layer.feature, element);
+  };
+
   const udpateSelection = () => {
     if(!mapInstance) return
     const { id, source } = queryStringParser(location.search);
     mapInstance.eachLayer((f: any) => {
       const isMarker = !f.getBounds;
       if (!f.feature) return;
       const featureId = String(isMarker ? f.feature.id : f.feature.properties.id);
-      if (featureId === id) {
-        const element = isMarker ? f._icon : f.getElement();
-        const bounds = isMarker ? [f.getLatLng(), f.getLatLng()] : f.getBounds();
-        mapInstance.fitBounds(bounds);
-        if (isMarker) {
-          mapInstance.setZoom(16);
-        }
-        onItemSelected(isMarker ? 'devices' : 'cameras', f.feature, element);
+      if (isMarker) {
```

Comment:
Does this cover the case when the marker is not inside a cluster on the default zoom level?

Reply:
clustering only happens with markers and nog areas

Reply:
I got that but if the marker is not in a cluster, this branch won't highlight it. Pease check 
- http://localhost:3001/?id=100&category=Camera&source=bfa 
- http://localhost:3001/?id=706&category=Baken&source=iothings
for highlight

Reply:
I fixed that issue with highlighting markers and areas